### PR TITLE
Feature/node worker full duplex 1

### PR DESCRIPTION
**Pull request by sydhds**

### CODE REVIEW

File: massa-network-worker/src/node_worker.rs [hunk 1]
```diff
@@ -161,14 +117,29 @@ impl NodeWorker {
                     }
                 };
             }
+
+            if let Err(e) = socket_writer.write_half.shutdown().await {
```
Comment (AurelienFT) [inline massa-network-worker/src/node_worker.rs:121]: Why we are shutting down the first writer ? We can't make one single writer for the whole process ?

**Review by AurelienFT — COMMENTED** [review]
Can you add debug print on critical cases of your feature ?

File: massa-network-worker/src/node_worker.rs (hunk 1 above)
Comment (sydhds) [reply massa-network-worker/src/node_worker.rs:121]: What do you mean by first writer? And not sure to understand your 2nd question too :-/

**Comment by sydhds** [comment]
> Can you add debug print on critical cases of your feature ?

done

File: massa-network-worker/src/node_worker.rs (hunk 1 above)
Comment (AurelienFT) [reply massa-network-worker/src/node_worker.rs:121]: Ok my bad

**Review by AurelienFT — APPROVED** [review]
The debug print should use debug macro massa_trace had to be reworked we never run node with this debug level.

**Review by AurelienFT — DISMISSED** [review]
Ok on my side. if you tested it also it's fullyyyyyy ok !

**Review by AurelienFT — COMMENTED** [review]
I think it would be nice to add documentation for all new functions.

**Comment by AurelienFT** [comment]
Becareful your branch is pointing to loop_timers

**Comment by damip** [comment]
@sydhds can you rebase the branch on main ?

**Comment by damip** [comment]
@sydhds  let's also do this:
```
NODE_SEND_CHANNEL_SIZE = 10_000
NETWORK_CONTROLLER_CHANNEL_SIZE = 10_000
NETWORK_EVENT_CHANNEL_SIZE = 10_000
NETWORK_NODE_COMMAND_CHANNEL_SIZE = 10_000
NETWORK_NODE_EVENT_CHANNEL_SIZE = 10_000
```

File: massa-consensus-worker/src/consensus_worker.rs [hunk 2]
```diff
@@ -501,6 +501,8 @@ impl ConsensusWorker {
                 slot,
                 storage,
             } => {
+                warn!("CONSENSUS received_block START");
```
Comment (damip) [inline massa-consensus-worker/src/consensus_worker.rs:504]: all those warnings are artifacts from loop_timers

File: massa-network-exports/src/commands.rs [hunk 3]
```diff
@@ -132,11 +132,41 @@ pub enum NodeEventType {
...
+        }
+    }
+}
+
 /// Events node worker can emit.
 /// Events are a tuple linking a node id to an event type
 #[derive(Clone, Debug)]
 pub struct NodeEvent(pub NodeId, pub NodeEventType);
 
+impl NodeEvent {
```
Comment (damip) [inline massa-network-exports/src/commands.rs:163]: all the get_info are artifacts from loop_timers

**Comment by AurelienFT** [comment]
@sydhds why there is a 2 PR ?

**Comment by damip** [comment]
> @sydhds why there is a 2 PR ?

This one has a lot of artifacts from loop_timers.
@sydhds created a new, clean one based on main with just his commits.

**Pull request closed without merging**